src/registrar.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const assert = require('assert')
+const debug = require('debug')
+const log = debug('libp2p:peer-store')
+log.error = debug('libp2p:peer-store:error')
+const errCode = require('err-code')
+
+const { Connection } = require('libp2p-interfaces/src/connection')
+const PeerInfo = require('peer-info')
+const MulticodecTopology = require('./connection-manager/topology')
+
+/**
+ * Responsible for notifying registered protocols of events in the network.
+ */
+class Registrar {
+  /**
+   * @param {PeerStore} peerStore
+   * @constructor
+   */
+  constructor (peerStore) {
```

Comment:
Let's pass an object, we may need to expand this in the future.
```suggestion
  constructor ({ peerStore }) {
```